File: src/ProjectR/content/scripts/spells/Ragnarok.lua
function GetTargetType()
  return Myself
end

function GetName()
  return "Ragnarok"
end

function GetDescription()
  return "Fires a concentrated laser of pure arcane energy at your enemies.\nUses up all of your MP, making the spell stronger.\nYou are completely exhausted after using this."
end

function IsSupportSpell()
    return false
end

function GetMPCost()
    return 1.6
end

function GetDelay()
  return 0
end

function GetSpellType()
  return Magical
end

function GetMasteries()
  return { ARC }
end

function SpellEffect()
  a:UseMP(160)
  damageMod = a:GetCurrentMP() / 10
  ForEachDefenderParty("DamageCalc")
  a:UseMP(200)
end

function DamageCalc(character)
  local charStats = GetStats(character)
  local defMR = charStats:GetTotalStat(MR)
  local defARC = charStats:GetTotalStat(ARC)

  local damage = ((8 + damageMod) * aMD * (aARC/100) - .5 * defMR) * (100/defARC)
  character:TakeDamage(damage)
end

File: src/ProjectR.Scripting/content/scripts/spells/Ragnarok.cs
using System.Collections.Generic;
using ProjectR.Interfaces;
using ProjectR.Interfaces.Model;
using ProjectR.Interfaces.Model.Stats;

namespace ProjectR.Scripting.Spells
{
    public class Ragnarok : SpellScriptBase
    {
        public override string Name { get { return "Ragnarok"; } }
        public override string Description { get { return "Fires a concentrated laser of pure arcane energy at your enemies.\nUses up all of your MP, making the spell stronger.\nYou are completely exhausted after using this."; } }

        public override TargetType TargetType { get { return TargetType.Enemies; } }
        public override IList<EleMastery> Masteries { get { return new EleMastery[] { EleMastery.ARC, }; } }
        public override SpellType SpellType { get { return SpellType.Magical; } }
        public override bool IsSupportSpell { get { return false; } }

        public override double MPCost { get { return 160; } }
        public override double Delay { get { return 0; } }

        public override void SpellEffect(ICharacter caster, IList<ICharacter> targets)
        {
            var damageMod = caster.CurrentMP / 10;
            foreach (var target in targets)
            {
                Target = target;
                SpellEffect(caster, target, damageMod);
            }
            caster.UseMP(200);
        }

        public override void SpellEffect(ICharacter caster, ICharacter target, double damageMod)
        {
            var damage = ((8 + damageMod) * aMD * (aARC / 100) - .5 * dMR) * (100 / dARC);
            DealDamage(damage);
        }
    }
}